File: src/main/java/io/dustin/finstage/financial/infra/client/SymbolClient.java
package io.dustin.finstage.financial.infra.client;

import io.dustin.finstage.financial.dto.ExternalSymbolListResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

public interface SymbolClient {
    ExternalSymbolListResponse fetchSymbols(int page, int size);
}

File: src/main/kotlin/io/dustin/finstage/financial/infra/client/SymbolClient.kt
package io.dustin.finstage.financial.infra.client

import io.dustin.finstage.common.annotation.ExternalClient
import io.dustin.finstage.common.http.HttpClient
import io.dustin.finstage.financial.dto.ExternalSymbolListResponse

@ExternalClient
class SymbolClient(
        private val httpClient: HttpClient
) {

    fun fetchSymbols(page: Int, size: Int): ExternalSymbolListResponse? {
        val url = "http://localhost:8081/symbols?page=$page&size=$size"
        return httpClient.get(url, ExternalSymbolListResponse::class.java)
    }
}
